UI/MemStickScreen.cpp

```diff
@@ -0,0 +1,190 @@
+	System_InputBoxGetString(sy->T("Memory Stick Folder"), g_Config.memStickDirectory, [&](bool result, const std::string &value) {
+		auto sy = GetI18NCategory("System");
+		auto di = GetI18NCategory("Dialog");
+
+		if (result) {
+			std::string newPath = value;
+			size_t pos = newPath.find_last_not_of("/");
+			// Gotta have at least something but a /, and also needs to start with a /.
+			if (newPath.empty() || pos == newPath.npos || newPath[0] != '/') {
+				settingInfo_->Show(sy->T("ChangingMemstickPathInvalid", "That path couldn't be used to save Memory Stick files."), nullptr);
+				return;
+			}
+			if (pos != newPath.size() - 1) {
+				newPath = newPath.substr(0, pos + 1);
+			}
+
+			pendingMemStickFolder_ = newPath;
+			std::string promptMessage = sy->T("ChangingMemstickPath", "Save games, save states, and other data will not be copied to this folder.\n\nChange the Memory Stick folder?");
+
+			//if (!File::Exists(newPath)) {
+			//	promptMessage = sy->T("ChangingMemstickPathNotExists", "That folder doesn't exist yet.\n\nSave games, save states, and other data will not be copied to this folder.\n\nCreate a new Memory Stick folder?");
+			//}
+			// Add the path for clarity and proper confirmation.
+			promptMessage += "\n\n" + newPath + "/";
+			screenManager()->push(new PromptScreen(promptMessage, di->T("Yes"), di->T("No"), std::bind(&MemStickScreen::CallbackMemStickFolder, this, std::placeholders::_1)));
+		}
+	});
+	*/
+	return UI::EVENT_DONE;
+}
+
+void MemStickScreen::sendMessage(const char *message, const char *value) {
+	// Always call the base class method first to handle the most common messages.
+	UIDialogScreenWithBackground::sendMessage(message, value);
+
+	if (screenManager()->topScreen() == this) {
+		if (!strcmp(message, "browse_folderSelect")) {
+			std::string filename;
+			filename = value;
+			INFO_LOG(SYSTEM, "Got folder: '%s'", filename.c_str());
+			// Don't allow browsing in the other tabs (I don't think it's possible to reach the option though)
+			pendingMemStickFolder_ = Path(filename);
+			CallbackMemStickFolder(true);
+		}
+	}
+}
+
+void MemStickScreen::CallbackMemStickFolder(bool yes) {
+	auto sy = GetI18NCategory("System");
+
+	if (yes) {
+		Path memStickDirFile = Path(g_Config.internalDataDirectory) / "memstick_dir.txt";
+		Path testWriteFile = Path(pendingMemStickFolder_) / ".write_verify_file";
+
+		// Doesn't already exist, create.
+		// Should this ever happen?
+		if (pendingMemStickFolder_.Type() == PathType::NATIVE) {
+			if (!File::Exists(pendingMemStickFolder_)) {
+				File::CreateFullPath(pendingMemStickFolder_);
+			}
+			if (!File::WriteDataToFile(true, "1", 1, testWriteFile)) {
+				settingInfo_->Show(sy->T("ChangingMemstickPathInvalid", "That path couldn't be used to save Memory Stick files."), nullptr);
+				return;
+			}
+			File::Delete(testWriteFile);
+		} else {
+			// TODO: Do the same but with scoped storage? Not really necessary, right? If it came from a browse
```

Comment:
Can we end up with a folder or SD card that is read only?

-[Unknown]

Reply:
We do ask for a writable folder, so hopefully not...